### ISSUE TITLE
Double args to JS functions in wasm32 not working

### ISSUE THREAD

**Issue by FloFri** [JetBrains/kotlin-native#1647]
There seems to be a problem with transferring double arguments to the JS side from within the wasm32 target. Every double value is 0 on the JS side.

I have traced down the error down to the doubleUpper and doubleLower functions.

Have a look at this code:

```
const val upperWord = 0xffffffff.toLong() shl 32

fun doubleUpper(value: Double): Int =
    ((value.toBits() and upperWord) ushr 32) .toInt()

fun doubleLower(value: Double): Int =
    (value.toBits() and 0x00000000ffffffff) .toInt()


fun main(args: Array<String>) {    
    val x: Double = 40.0;
    println(doubleUpper(x));
    println(doubleLower(x));
}
```

The top part is copied from Interop/JsRuntime/src/main/kotlin/jsinterop.kt for easier testing.
The main problem here is that upperWord = 0 after the shift but I don't know exactly why. Maybe because wasm is 32 bit only?

**Comment by alexander-gorshenev** [comment]
There seems to be an issue with the order of globals initialization. I will take a look.

**Comment by FloFri** [comment]
@alexander-gorshenev Browser (Chromium 64.0.3282.140) with Kotlin-Native version 0.7

**Comment by olonho** [comment]
Does https://github.com/JetBrains/kotlin-native/pull/1654 help?

**Comment by alexander-gorshenev** [comment]
@FloFri Should have been fixed by #1654

**Comment by FloFri** [comment]
Thanks. That fixed it!
